src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -605,6 +613,228 @@ void copy_columns_by_row_chunks(const ov::Tensor& src, ov::Tensor& dst) {
 namespace ov {
 namespace genai {
 
+SMStaticLLMPipeline::SMStaticLLMPipeline(
+    const std::filesystem::path& models_path,
+    const ov::genai::Tokenizer& tokenizer,
+    const std::string& device,
+    const ov::AnyMap& config
+) : LLMPipelineImplBase(tokenizer,
+                        utils::from_config_json_if_exists(models_path)) {
+    ov::Core core;
+    auto model = core.read_model((models_path / "openvino_model.xml").string());
+
+    ov::AnyMap properties = config;
+    const uint32_t kMaxPromptLen = pop_int_and_cast(properties, "MAX_PROMPT_LEN").value_or(1024u);
+    const uint32_t kMinResponseLen = pop_int_and_cast(properties, "MIN_RESPONSE_LEN").value_or(128u);
+    std::string generate_hint = pop_or_default<std::string>(properties, "GENERATE_HINT", "FAST_COMPILE");
+    ModelDesc model_desc = get_modeldesc_from_json(models_path / "config.json");
+
+    if (properties.count("NPU_USE_NPUW") == 0u) {
+        properties.insert({{"NPU_USE_NPUW", "YES"}});
+    }
+    if (properties.count("NPUW_LLM") == 0u) {
+        properties.insert({{"NPUW_LLM", "YES"}});
+    }
+
+    properties.insert({{"NPUW_LLM_MODEL_DESC", model_desc_to_map(model_desc)}});
+    properties.insert({{"NPUW_LLM_MAX_PROMPT_LEN", kMaxPromptLen}});
+    properties.insert({{"NPUW_LLM_MIN_RESPONSE_LEN", kMinResponseLen}});
+    properties.insert({{"NPUW_LLM_GENERATE_HINT", generate_hint}});
+    auto compiled = core.compile_model(model, "NPU", properties);
+
+    m_request  = compiled.create_infer_request();
+}
+
+DecodedResults SMStaticLLMPipeline::generate(
+    StringInputs inputs,
+    OptionalGenerationConfig generation_config,
+    StreamerVariant streamer
+) {
+    auto start_time = std::chrono::steady_clock::now();
+
+    GenerationConfig config = (generation_config.has_value()) ? *generation_config : m_generation_config;
+    std::string prompt;
+    if (auto input_vector = std::get_if<std::vector<std::string>>(&inputs)) {
+        if (input_vector->size() > 1u) {
+            OPENVINO_THROW("Currently only batch size=1 is supported");
+        }
+        OPENVINO_ASSERT(!input_vector->empty());
+        prompt = std::move(input_vector->front());
+    } else {
+        OPENVINO_ASSERT(std::holds_alternative<std::string>(inputs));
+        prompt = std::get<std::string>(inputs);
+    }
+
+    ov::genai::TokenizedInputs tokenized_input;
+    //if (m_is_chat_conversation) {
```

Comment:
Why is it commented out?

Reply:
It should be implemented in future, added fixme, removed comments